src/test/unit_tests/test_s3_ops.js

```diff
@@ -751,6 +752,52 @@ mocha.describe('s3_ops', function() {
 
         });
 
+        mocha.it('should be able to set versioning', async function() {
+            if (!is_s3_namespace) this.skip();
+            this.timeout(600000);
+            await s3.putBucketVersioning({
+                Bucket: bucket_name,
+                VersioningConfiguration: {Status: "Enabled"}
+            });
+
+            let res = await s3.getBucketVersioning({Bucket: bucket_name});
+            assert.strictEqual(res.Status, "Enabled");
+
+            await s3.putBucketVersioning({
+                Bucket: bucket_name,
+                VersioningConfiguration: {Status: "Suspended"}
+            });
+            res = await s3.getBucketVersioning({Bucket: bucket_name});
+            assert.strictEqual(res.Status, "Suspended");
+        });
+
+        mocha.it('should be able to put/get object by version', async function() {
+            if (!is_s3_namespace) this.skip();
+            this.timeout(600000);
+            await s3.putBucketVersioning({
+                Bucket: bucket_name,
+                VersioningConfiguration: {Status: "Enabled"}
+            });
+
+            const put_res = await s3.putObject({
+                Bucket: bucket_name,
+                Key: text_file1,
+                Body: file_body2,
+                ContentType: 'text/plain'
+            });
+
+            //should work, object exist
+            await s3.getObject({ Bucket: bucket_name, Key: text_file1, VersionId: put_res.VersionId});
+
+            // clean up versioning products
+            await s3.deleteObject({ Bucket: bucket_name, Key: text_file1, VersionId: put_res.VersionId});
+
+            await s3.putBucketVersioning({
+                Bucket: bucket_name,
+                VersioningConfiguration: {Status: "Suspended"}
+            });
+        });
```

Comment:
After this step, you should compleatly delete the bucket and create it again.
Suspended versioned bucket is not the same as not enabled one.
This is changing the intention of the entire suite.